kernel/kernel-api/src/main/java/io/delta/kernel/internal/ActionCommitLog.java

```diff
@@ -0,0 +1,272 @@
+      throw new UncheckedIOException("Unable to close resource", e);
+    }
+    return output;
+  }
+
+  /**
+   * For a table get the list of commit log files for the provided version range.
+   *
+   * @param tablePath path for the given table
+   * @param startVersion start version of the range (inclusive)
+   * @param endVersion end version of the range (inclusive)
+   * @return the list of commit files in increasing order between startVersion and endVersion
+   * @throws TableNotFoundException if the table does not exist or if it is not a delta table
+   * @throws KernelException if a commit file does not exist for any of the versions in the provided
+   *     range
+   * @throws KernelException if provided an invalid version range
+   */
+  public static List<FileStatus> getCommitFilesForVersionRange(
+      Engine engine, Path tablePath, long startVersion, long endVersion) {
+
+    // Validate arguments
+    if (startVersion < 0 || endVersion < startVersion) {
+      throw invalidVersionRange(startVersion, endVersion);
+    }
+
+    // Get any available commit files within the version range
+    List<FileStatus> commitFiles = listCommitFiles(engine, tablePath, startVersion, endVersion);
+
+    // There are no available commit files within the version range.
+    // This can be due to (1) an empty directory, (2) no valid delta files in the directory,
+    // (3) only delta files less than startVersion prefix (4) only delta files after endVersion
+    if (commitFiles.isEmpty()) {
+      throw noCommitFilesFoundForVersionRange(tablePath.toString(), startVersion, endVersion);
+    }
+
+    // Verify commit files found
+    // (check that they are continuous and start with startVersion and end with endVersion)
+    verifyDeltaVersions(commitFiles, startVersion, endVersion, tablePath);
+
+    return commitFiles;
+  }
+
+  /**
+   * Read the given commitFiles and return the contents as an iterator of batches. Also adds two
+   * columns "version" and "timestamp" that store the commit version and timestamp for the commit
+   * file that the batch was read from.
+   *
+   * @param commitFiles list of delta commit files to read
+   * @param readSchema JSON schema to read
+   * @return an iterator over the contents of the files in the same order as the provided files
+   */
+  public static CloseableIterator<ColumnarBatch> readCommitFiles(
+      Engine engine, List<FileStatus> commitFiles, StructType readSchema) {
+
+    return new ActionsIterator(engine, commitFiles, readSchema, Optional.empty())
+        .map(
+            actionWrapper -> {
+              long timestamp =
+                  actionWrapper
+                      .getTimestamp()
+                      .orElseThrow(
+                          () ->
+                              new RuntimeException("Commit files should always have a timestamp"));
+              ExpressionEvaluator commitVersionGenerator =
+                  wrapEngineException(
+                      () ->
```

Comment:
nit: can this indentation not be this instead?
```java
() -> engine
    .getExpressionHandler()
    ...
```

this would help reduce the indentation level, which would help improve the readability

Reply:
`javafmtAll` reverts this change